File: pset1/greedy/greedy.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>


int main(void){
    float dollars;
    int cents;
    int coins = 0;

    printf("O hai! How much change is owed?\n");
    dollars = get_float();
    if(dollars <= 0){
        do{
            printf("How much change is owed?\n");
            dollars = get_float();
        }while(dollars <= 0);
    }

    cents = round(dollars * 100);

    while(cents >= 25){
        cents -= 25;
        coins++;
    }
    while(cents >= 10){
        cents -= 10;
        coins++;
    }
    while(cents >= 5){
        cents -= 5;
        coins++;
    }
    while(cents >= 1){
        cents -= 1;
        coins++;
    }

    printf("%i\n", coins);
}

File: pset6/greedy.py
# import modules
import cs50


# main
def main():
    # get dollars
    while True:
        print("O hai! How much change is owed?")
        d = cs50.get_float()
        if d > 0:
            break

    # get cents
    c = int(d * 100)

    # init coins
    coins = 0

    # calculate...
    coins += c // 25
    c %= 25
    coins += c // 10
    c %= 10
    coins += c // 5
    c %= 5
    coins += c

    # print results
    print("{}".format(coins))

    # done
    exit(0)


# redirect to main
if __name__ == "__main__":
    main()
